PDK_Abstraction/FinFET14nm_Mock_PDK/primitive.py

```diff
@@ -92,6 +92,23 @@ def _get_wire_terminators(intersecting_tracks):
                         minx, maxx = _get_wire_terminators([*locs, current_track])
                         self.addWire(self.m2, net, None, i, (minx, -1), (maxx, 1))
 
+    def _bodyContact(self, x, y, x_cells, name='M1'):
+        h = self.m2PerUnitCell
+        gu = self.gatesPerUnitCell
+        gate_x = x*gu + gu // 2
+        self._xpins[name]['B'].append(gate_x)
+        fullname = f'{name}_X{x}_Y{y}'
+        self.addWire( self.activeb, None, None, y, (x,1), (x+1,-1))
+        self.addWire( self.pb, None, None, y, (x,1), (x+1,-1)) 
+        self.addWire( self.m1, None, None, gate_x, ((y+1)*h+3, -1), ((y+1)*h+self.lFin//2-3, 1))
+        self.addWire( self.LISDb, None, None, gate_x, ((y+1)*h+3, -1), ((y+1)*h+self.lFin//2-3, 1)) 
+        self.addVia( self.va, f'{fullname}:B', None, gate_x, ((y+1)*h//2, self.lFin//4))
+        self.addVia( self.v1, f'{fullname}', None, gate_x, ((y+1)*h//2, self.lFin//4))
```

Comment:
The self.v1 line looks wrong. You are missing a pin name. You probably wanted to say f'{fullname}:B' just like the self.va connection?